lib/trino-orc/src/main/java/io/trino/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -479,6 +479,10 @@ private static List<Integer> createSliceColumnPositionList(
     public List<StreamDataOutput> getBloomFilters(CompressedMetadataWriter metadataWriter)
             throws IOException
     {
+        if (directEncoded) {
```

Comment:
Is there a way to test it?

How it was broken? (improve commie message)

Reply:
https://github.com/trinodb/trino/blob/2df56a057aedf3a59088212214ac84e7ce759f59/lib/trino-orc/src/main/java/io/trino/orc/OrcWriter.java#L386-L406

`SliceDictionaryColumnWriter#tryConvertToDirect` will clear rowGroups finally in dictionary compression optimization, index streams use the rowGroups after optimization that result in losing bloom filter streams